File: DeviantArtFs/Types/Wrappers/DeviantArtCategory.fs
namespace DeviantArtFs

open FSharp.Json

type IBclDeviantArtCategory =
    abstract member Catpath: string
    abstract member Title: string
    abstract member HasSubcategory: bool
    abstract member ParentCatpath: string

type DeviantArtCategory = {
    catpath: string
    title: string
    has_subcategory: bool
    parent_catpath: string
} with
    interface IBclDeviantArtCategory with
        member this.Catpath = this.catpath
        member this.Title = this.title
        member this.HasSubcategory = this.has_subcategory
        member this.ParentCatpath = this.parent_catpath

type internal DeviantArtCategoryList = {
    categories: DeviantArtCategory[]
} with
    static member ParseSeq json =
        let o = Json.deserialize<DeviantArtCategoryList> json
        o.categories :> seq<DeviantArtCategory>
    interface seq<DeviantArtCategory> with
        member this.GetEnumerator() = (this.categories :> seq<DeviantArtCategory>).GetEnumerator()
        member this.GetEnumerator() = this.categories.GetEnumerator()

File: DeviantArtFs.Types/DeviantArtCategory.cs
using System;

namespace DeviantArtFs.Types {
    public class DeviantArtCategory {
        public string catpath { get; set; } = "";
        public string title { get; set; } = "";
        public bool has_subcategory { get; set; }
        public string parent_catpath { get; set; } = "";
    }

    public class DeviantArtCategoryList {
        public DeviantArtCategory[] categories { get; set; } = new DeviantArtCategory[0];
    }
}
